normandy/base/api/mixins.py

```diff
@@ -12,3 +15,32 @@ def retrieve(self, request, *args, **kwargs):
     @cache_control(public=True, max_age=settings.API_CACHE_TIME)
     def list(self, request, *args, **kwargs):
         return super().list(request, *args, **kwargs)
+
+
+_midair_collision_enabled = True
+
+
+def _enable_midair_collision(enabled=True):
+    global _midair_collision_enabled
+    _midair_collision_enabled = enabled
+
+
+class MidairCollisionViewsetMixin(object):
+    """Modify a ModelViewSet to add midair collision to update methods"""
+
+    def update(self, request, *args, **kwargs):
+        if _midair_collision_enabled:
+            # Using a serializer gets some easy validation
+            serializer = LastUpdatedSerializer()
+            request_ts = serializer.to_internal_value(request.data)['last_updated']
```

Comment:
Instead of `to_internal_value`, could you pass `request.data` as the `data` kwarg to the serializer to deserialize it? http://www.django-rest-framework.org/api-guide/serializers/#deserializing-objects